pygmt/src/grdcontour.py

```diff
@@ -96,32 +100,45 @@ def grdcontour(self, grid, **kwargs):
     Example
     -------
     >>> import pygmt
-    >>> # load the 15 arc-minutes grid with "gridline" registration
-    >>> # in a specified region
+    >>> # Load the 15 arc-minutes grid with "gridline" registration in the
+    >>> # specified region
     >>> grid = pygmt.datasets.load_earth_relief(
     ...     resolution="15m",
     ...     region=[-92.5, -82.5, -3, 7],
     ...     registration="gridline",
     ... )
-    >>> # create a new plot with pygmt.Figure()
+    >>> # Create a new plot with pygmt.Figure()
     >>> fig = pygmt.Figure()
-    >>> # create the contour plot
+    >>> # Create the contour plot
     >>> fig.grdcontour(
-    ...     # pass in the grid downloaded above
+    ...     # Pass in the grid downloaded above
     ...     grid=grid,
-    ...     # set the interval for contour lines at 250 meters
+    ...     # Set the interval for contour lines at 250 meters
     ...     interval=250,
-    ...     # set the interval for annotated contour lines at 1,000 meters
+    ...     # Set the interval for annotated contour lines at 1,000 meters
     ...     annotation=1000,
-    ...     # add a frame for the plot
+    ...     # Add a frame for the plot
     ...     frame="a",
-    ...     # set the projection to Mercator for the 10 cm figure
+    ...     # Set the projection to Mercator for the 10 cm figure
     ...     projection="M10c",
     ... )
-    >>> # show the plot
+    >>> # Show the plot
     >>> fig.show()
     """
     kwargs = self._preprocess(**kwargs)
+
+    if is_nonstr_iter(kwargs.get("A")):
+        if len(kwargs["A"]) == 1:
+            kwargs["A"] = str(kwargs["A"][0]) + ","
+        else:
+            kwargs["A"] = ",".join(f"{item}" for item in kwargs["A"])
+
+    if is_nonstr_iter(kwargs.get("C")):
+        if len(kwargs["C"]) == 1:
+            kwargs["C"] = str(kwargs["C"][0]) + ","
+        else:
+            kwargs["C"] = ",".join(f"{item}" for item in kwargs["C"])
```

Comment:
These two code blocks have exactly the same code, except one for `A` and another for `C`. Please rewrite it using a for-loop instead, e.g., 
```
for arg in ["A", "C"]:
    if ...

```

Reply:
Totally right, done in the commits https://github.com/GenericMappingTools/pygmt/pull/3116/commits/c8772bff9998b865046efb5ba416b1b1abec7962 and https://github.com/GenericMappingTools/pygmt/pull/3116/commits/9a9ae495aaacdc8d69db7944eea39177cb127bb7.